src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/RedundantImportCheckTest.java

```diff
@@ -50,6 +57,22 @@ public void testGetRequiredTokens() {
     }
 
     @Test
+    @SuppressWarnings("unchecked")
+    public void testStateIsClearedOnBeginTree() throws Exception {
+        final RedundantImportCheck check = new RedundantImportCheck();
+        final File input = new File(getPath("InputRedundantImportCheckClearState.java"));
+        final DetailAST importAst = TestUtils.parseFile(input).getNextSibling();
+        final DetailAST staticImportAst = importAst.getNextSibling();
+
+        assertTrue("Stateful field is not cleared after beginTree",
+            TestUtils.isStatefulFieldClearedDuringBeginTree(check, importAst, "imports",
+                imports -> ((Collection<FullIdent>) imports).isEmpty()));
+        assertTrue("Stateful field is not cleared after beginTree",
+            TestUtils.isStatefulFieldClearedDuringBeginTree(check, staticImportAst,
+                "staticImports", imports -> ((Collection<FullIdent>) imports).isEmpty()));
```

Comment:
I need good reason why this can not be tested by `verify`.
Usage of `isStatefulFieldClearedDuringBeginTree` is ok , only when you prove that it is impossible to cover is logic is working properly.